packages/api/src/external/carequality/shared.ts

```diff
@@ -19,14 +19,22 @@ export function isGWValid(gateway: { homeCommunityId: string; url: string }): bo
 }
 
 export async function validateCQEnabledAndInitGW(
-  cxId: string,
+  patient: Pick<Patient, "id" | "cxId">,
+  facilityId: string,
   forceEnabled: boolean,
   outerLog: typeof console.log
 ): Promise<IHEGateway | undefined> {
+  const { cxId } = patient;
+
   try {
     const iheGateway = makeIheGatewayAPIForPatientDiscovery();
     const isCQEnabled = await isCarequalityEnabled();
     const isCQDirectEnabled = await isCQDirectEnabledForCx(cxId);
+    const isCqQueryEnabled = await isHieEnabledToQuery(
```

Comment:
nit: we can have 2 funcs `isFacilityEnabledToQueryCQ()` & `isFacilityEnabledToQueryCW()` which use `isHieEnabledToQuery` under the good, and use those respectively, instead of passing ` MedicalDataSource` everywhere